numpy/_core/defchararray.py

```diff
@@ -1385,17 +1385,19 @@ def replace(a, old, new, count=None):
     """
     a_arr = numpy.asarray(a)
     max_int64 = numpy.iinfo(numpy.int64).max
-    count = count if count is not None else max_int64
-
     counts = numpy._core.umath.count(a_arr, old, 0, max_int64)
+    if count is not None:
+        if numpy.any(count < 0):
+            raise ValueError("count must be positive or zero.")
```

Comment:
This is not correct. Negative values signify that all occurences should be changed, so `count[count < 0] = max_int64` is the way to go. I'm currently working on a slight rewrite of `replace`, which'll fix this on the C side.

Reply:
With my change here, it will need to be fixed here too, since the `counts` array is now adjusted (which is necessary since otherwise string lengths would be calculated incorrectly).

Reply:
I think adjusting here is much easier - just `counts = np.minimum(counts, count, where=count >= 0, out=counts)`